### PR TITLE
[clang][docs] Improve "Obtaining Clang" section

### PR DESCRIPTION
**Pull request by robincaloudis**
## Why

The documentation is written relatively to `clang-llvm`, not the root repository directory. However, some steps in the documentation are relative to the repository root, which is not correct.

## What
Documentation steps have been modified to make them correct and outdated ones were updated. Some details:
* Correct paths in documentation
* Change `bootstrap.py` -> `configure.py` since bootstraping Ninja has [slightly changed](https://github.com/ninja-build/ninja/tree/master?tab=readme-ov-file#python)

### CODE REVIEW

**Comment by robincaloudis** [comment]
@r4nt, could you please review? Thank you.

**Comment by robincaloudis** [comment]
Ping

**Comment by robincaloudis** [comment]
Ping.

**Comment by robincaloudis** [comment]
Ping

**Comment by Endilll** [comment]
LGTM, but wair for other reviewers.

File: clang/docs/LibASTMatchersTutorial.rst
Comment (jrtc27) [inline clang/docs/LibASTMatchersTutorial.rst:26]: Given the:
```
      cd ~/clang-llvm
      git clone git://cmake.org/stage/cmake.git
```
this isn't a good idea. Better to clone it to an llvm-project subdirectory (i.e. drop the .) and alter later steps.

Comment (robincaloudis) [reply clang/docs/LibASTMatchersTutorial.rst]: Thanks for the review @jrtc27. Indeed, having the `llvm-project` subdirectory is much cleaner. I updated all required steps. Found some more outdated details. Please re-review. Thank you.

**Review by jrtc27 — APPROVED** [review]
Thanks, this looks better now

**Comment by jrtc27** [comment]
Though please make sure to update the PR message (which will become the commit message) before merging, as it's no longer accurate

**Comment by robincaloudis** [comment]
> Though please make sure to update the PR message (which will become the commit message) before merging, as it's no longer accurate

Thanks. Description is updated 👍

**Comment by robincaloudis** [comment]
@jrtc27, I do not have write access. All checks passed. Can you merge the PR? Thanks.

**Pull request merged**
